### ISSUE TITLE
Kismatic fails with non-useful error when SSH key has wrong permissions

### ISSUE THREAD

**Issue by eriklg** [apprenda/kismatic#138]
I have run into various SSH-related issues during the validation phase, and the error message returned by kismatic is just "UNREACHABLE".

This makes me think that the node is unreachable on the network. My first instinct is to ping the machines, and once that passes I am out of ideas.

After some investigation, I found that the issue was related to the SSH key having wrong permissions.

**Comment by dasmb** [comment]
Yes I agree this is frustrating; I've run in to it myself on a few occasions. At the moment we're relying on Ansible to deliver this error message and it has a tendency to treat a whole host of SSH problems as "unreachable," with the real reason hidden in a stream of verbose logging.

And to be fair, there are lots of reasons why SSH can fail, but we could probably identify some of the top culprits (bad key negotiation, wrong host, username rejected). What was your actual SSH issue?

If you don't mind, could you give me a quick walk through of how you discovered the problem? I would love to help improve the debug process as well.

**Comment by eriklg** [comment]
The error in this case was due to wrong permissions on the ssh key file (permissions were too broad). 

The troubleshooting steps I took were:
1. Ping the machines (success)
2. Reading the documentation, it does claim that SSH errors will result in an unreachable error so I tried to log in to one of the nodes from the machine running the installer using the key and that's when I saw the actual error of the permissions.
